src/src/Commands/GPIO.cpp

```diff
@@ -0,0 +1,392 @@
+	  String dummy;
+	  SendStatusOnlyIfNeeded(event->Source, sendStatusFlag, key, dummy, 0);
+    return return_command_success();
+  } else {
+    return return_command_failed();
+  }
+}
+
+String Command_GPIO_Pulse(struct EventStruct *event, const char* Line)
+{
+  String logPrefix;
+  bool caseFound = true;
+  byte pluginID=0;
+  switch (Line[0])
+  {
+    case 'p': // pulse or pcfpulse
+      if (Line[1]=='u') { //pulse
+        pluginID=PLUGIN_GPIO;
+        logPrefix=String(F("GPIO"));
+      } else if (Line[1]=='c'){ //pcfpulse
+        pluginID=PLUGIN_PCF;
+        logPrefix=String(F("PCF"));
+      }
+      break;
+    case 'm': //mcp
+      pluginID=PLUGIN_MCP;
+      logPrefix=String(F("MCP"));
+      break;
+    default:
+      caseFound=false;
+  }
+
+  if (caseFound && checkValidPortRange(pluginID, event->Par1))
+  {
+    const uint32_t key = createKey(pluginID,event->Par1);
+
+    createAndSetPortStatus_Mode_State(key,PIN_MODE_OUTPUT,event->Par2);
+    GPIO_Write(pluginID, event->Par1, event->Par2);
+
+    delay(event->Par3);
+
+    createAndSetPortStatus_Mode_State(key,PIN_MODE_OUTPUT,!event->Par2);
+    GPIO_Write(pluginID, event->Par1, !event->Par2);
+
+    String log = logPrefix + String(F(" : port ")) + String(event->Par1);
+    log += String(F(". Pulse set for ")) + String(event->Par3)+String(F(" ms"));
+    addLog(LOG_LEVEL_INFO, log);
+    SendStatusOnlyIfNeeded(event->Source, SEARCH_PIN_STATE, key, log, 0);
+
+    return return_command_success();
+  } else {
+    logErrorGpioOutOfRange(logPrefix,event->Par1);
+    return return_command_failed();
+  }
+}
+
+String Command_GPIO_Toggle(struct EventStruct *event, const char* Line)
+{
+  String logPrefix;
+  byte pluginID;
+  //Line[0]='g':gpiotoggle; ='p':pcfgpiotoggle; ='m':mcpgpiotoggle
+  bool success = getPluginIDAndPrefix(Line[0], &pluginID, &logPrefix);
+  if (success && checkValidPortRange(pluginID, event->Par1))
+  {
+    const uint32_t key = createKey(pluginID,event->Par1);
+    // WARNING: operator [] creates an entry in the map if key does not exist
+    // So the next command should be part of each command:
+    byte mode;
+    int8_t state;
+
+    if (existPortStatus(key))
+    {
+      mode=globalMapPortStatus.at(key).mode;
+      state=globalMapPortStatus.at(key).state;
+    } else {
+      GPIO_Read(pluginID, event->Par1, &state);
+      mode = (state==-1)?PIN_MODE_OFFLINE:PIN_MODE_OUTPUT;
+    }
+
+    switch (mode) {
+      case PIN_MODE_OUTPUT:
+      case PIN_MODE_UNDEFINED:
+        {
+          createAndSetPortStatus_Mode_State(key,PIN_MODE_OUTPUT,!state);
+          GPIO_Write(pluginID, event->Par1, !state);
+
+          String log = logPrefix + String(F(" toggle: port#")) + String(event->Par1) + String(F(": set to ")) + String(!state);
+          addLog(LOG_LEVEL_ERROR, log);
+      	  SendStatusOnlyIfNeeded(event->Source, SEARCH_PIN_STATE, key, log, 0);
+
+          return return_command_success();
+        }
+        break;
```

Comment:
Has this been taken from older code?
I thought I had changed these break-after-return statements a while ago (not entirely sure though)

Reply:
I have rebased my fork a week ago. So this should be recent code.
But the lines you are pointing to are my code. 
How should I change it?

Reply:
I remember the static code analyzer tools I used were warning me about having a break after a return statement, which is unreachable code and often not as intended.